dialer.go

```diff
@@ -565,3 +597,43 @@ func (d *Dialer) connectionInfoCache(
 
 	return c, nil
 }
+
+func (d *Dialer) queryDNS(ctx context.Context, domainName string) (instance.ConnName, error) {
+	// Attempt to query the SRV records.
+	// This could return a partial error where both err != nil && len(records) > 0.
+	_, records, err := d.resolver.LookupSRV(ctx, "", "", domainName)
+
+	// Process the records returning the first valid SRV record.
+
+	// Sort the record slice so that lowest priority comes first.
+	sort.Slice(records, func(i, j int) bool {
+		return records[i].Priority < records[j].Priority
+	})
+	var perr error
+	// Attempt to parse records, returning the first valid record.
+	for _, record := range records {
+		// Remove trailing '.' from target value.
+		target := strings.TrimRight(record.Target, ".")
+
+		// Parse the target as a CN
+		cn, parseErr := instance.ParseConnName(target)
+		if parseErr != nil {
+			perr = fmt.Errorf("unable to parse SRV for %s: %v", domainName, parseErr)
+			continue
+		}
+		return cn, nil
+	}
+
+	// If resolve failed and no records were found, return the error.
+	if err != nil {
+		return instance.ConnName{}, fmt.Errorf("unable to resolve SRV record for %s: %v", domainName, err)
```

Comment:
%q might be nice here for the SRV record -- it will wrap the value in quotes.

Reply:
Fixed.